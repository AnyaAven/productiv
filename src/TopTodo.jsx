import Todo from "./Todo.jsx";

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo
 */

function TopTodo({ todos }) {
  // lowest-priority # is the highest priority
  const top = todos.reduce(
    (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);

  return <Todo
    id={top.id}
    title={top.title}
    description={top.description}
    priority={top.priority} />;
}

export default TopTodo;