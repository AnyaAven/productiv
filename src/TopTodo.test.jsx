import { test, expect, describe } from "vitest";
import { render, fireEvent } from "@testing-library/react";
import TopTodo from "./TopTodo.jsx";
import { TODOS } from "./_testCommon.js";

describe("smoke tests", function () {
  test("renders without crashing", function () {
    render(<TopTodo />);
  });
  test("matches snapshot on initial render", function () {
    const { container } = render(<TopTodo />);
    expect(container).toMatchSnapshot();
  });
});

describe("works", function () {
  test("todo is number 1 priority", function () {
    const { container } = render(<TopTodo todos={TODOS}/>);

    const qs = container.querySelector.bind(container);
    const qsa = container.querySelectorAll.bind(container);

    expect(qsa(".Todo").length).toEqual(1);
    expect(qs(".Todo")).toHaveTextContent("priority: 1");
  });

});