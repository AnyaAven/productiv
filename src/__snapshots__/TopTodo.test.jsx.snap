// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`smoke tests > matches snapshot on initial render 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b />
       
      <small>
        (priority: 
        )
      </small>
    </div>
    <div>
      <small />
    </div>
  </div>
</div>
`;
